lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReader.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import org.apache.parquet.internal.filter2.columnindex.RowRanges;
+
+public interface ColumnReader
+{
+    boolean hasPageReader();
```

Comment:
nit: how could it not have page reader? Why wouldn't caller track it?

Reply:
It's used in the existing ParquetReader code to figure out when to create page reader and give it to column reader. I think the page reader creation and hand-off to column reader takes places lazily rather than on initialisation of column reader to allow skipping some work if lazy loading of block helps to eliminate read of some column.
In any case, considering changes to that is out of scope here.